### ISSUE TITLE
Animation always plays at least twice

### ISSUE THREAD

**Issue by thijs-feel** [Gamote/lottie-react#48]
**Describe the bug**
When playing an animation the first loop is not 'registered' by the library. 

**To Reproduce**
1. Create a fresh react app using create-react-app and install lottie-react
2. Add a Lottie component to the app and subscribe to the onLoopComplete callback
3. Start the server and play the animation
4. Error: Only starting from the second time the animation plays the onLoopComplete callback is called.
5. Set the Lottie component to loop=false
6. Play the animation
7. Error: Animation plays twice

**Expected behavior**
Animation plays only once when loop is set to false.
Animation calls onLoopComplete after every loop.

**Desktop (please complete the following information):**
 - OS: Macos 11.6 (Big Sur)
 - Browser: Chrome, Firefox, Safari
 - Version: 2.2.1

Workaround I'm using currently:
Play the animation with the playSegments function with [0, total_frames]

**Comment by MilanPoznan** [comment]
> **Describe the bug** When playing an animation the first loop is not 'registered' by the library.
> 
> **To Reproduce** Steps to reproduce the behavior:
> 
> 1. Create a fresh react app using create-react-app and install lottie-react
> 2. Add a Lottie component to the app and subscribe to the onLoopComplete callback
> 3. Start the server and play the animation
> 4. Error: Only starting from the second time the animation plays the onLoopComplete callback is called.
> 5. Set the Lottie component to loop=false
> 6. Play the animation
> 7. Error: Animation plays twice
> 
> **Expected behavior** Animation plays only once when loop is set to false. Animation calls onLoopComplete after every loop.
> 
> **Desktop (please complete the following information):**
> 
> * OS: Macos 11.6 (Big Sur)
> * Browser: Chrome, Firefox, Safari
> * Version: 2.2.1
> 
> Workaround I'm using currently: Play the animation with the playSegments function with [0, total_frames]

I had the same issue, you can use previous version 2.1. So in package.json just remove ^ so you can set 
 "lottie-react": "2.1.0", just without ^. 
 Hope that this will help

**Comment by Gamote** [comment]
Hey @thijs-feel and @MilanPoznan for the input. I will have a look.

**Comment by thijs-feel** [comment]
I can confirm that the bug does not seem to happen on 2.1.0 @MilanPoznan Thanks for the tip!

**Comment by MilanPoznan** [comment]
@thijs-feel Cool, no problem!

**Comment by drager** [comment]
Happens for me in `2.2.x` as well. `2.1.0` works fine.

**Comment by raimohanska** [comment]
Stumbled upon this issue as well. Here's a [Code sandbox](https://codesandbox.io/s/lottie-animation-example-15i53?file=/src/App.tsx) for reproduction, in case it helps. I reverted to 2.1.0 as a workaround for now.

Thanks for the great library!

**Comment by Gamote** [comment]
Thank you @raimohanska for the example, very helpful. It will be fixed in the next release.

**Comment by hanayashiki** [comment]
when will you do the next release?

**Comment by arnoutj** [comment]
Happened to us as well and we can't revert to `2.1.0` as we are using React version 17

**Comment by jcontonio** [comment]
> Happened to us as well and we can't revert to `2.1.0` as we are using React version 17

Same here... can't revert, it's playing twice even if I try to keep track of how many times it's played in a ref.

**Comment by johnchourajr** [comment]
+1 on this bug, having the same issue.

**Comment by ndcollins** [comment]
+1 on this bug. React 18 gives the following warning:

Warning: componentWillUpdate has been renamed, and is not recommended for use. See https://reactjs.org/link/unsafe-component-lifecycles for details.

* Move data fetching code or side effects to componentDidUpdate.
* Rename componentWillUpdate to UNSAFE_componentWillUpdate to suppress this warning in non-strict mode. In React 18.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: Lottie

**Comment by draperj** [comment]
Ran into this issue as well. Occurs on first play but not subsequent plays on ver 2.2.1. Confirmed fine in 2.1.0.

**Comment by clempat** [comment]
I was taking a look into this issue and saw that this happening due to resetting the segment there: https://github.com/Gamote/lottie-react/blob/main/src/hooks/useLottie.tsx#L224

I am not sure what would be the implication of not doing this reset. But the animation run once, then there is this reset then the animation run a second time.

**Comment by lucasrivoiro** [comment]
+1 on this bug. Please, make a fix for this bug. I cant revert to 2.1.0 because my react lib is in version 17

**Comment by usercao** [comment]
+1 on this issue

```json
"lottie-react": "^2.3.0",
"react": "^17.0.2",
```

@Gamote  When is the problem expected to be fixed? The problem is very serious.

**Comment by AhsanSarwar45** [comment]
➕ on this bug.
```json
"react": "18.1.0",
"lottie-react": "^2.3.0",
```

**Comment by carl0shd** [comment]
+1

**Comment by jgalianoz** [comment]
+1

**Comment by Gamote** [comment]
This issue was fixed in [`v2.3.1`](https://github.com/Gamote/lottie-react/releases/tag/v2.3.1).
